src/all/novelfull/src/eu/kanade/tachiyomi/extension/all/novelfull/NovelFullGlobal.kt

```diff
@@ -0,0 +1,256 @@
+    }
+
+    override fun fetchPopularManga(page: Int): Observable<MangasPage> {
+        if (page == 1) {
+            val url = "$baseUrl/comic/index.html?language=$lang"
+            fetchGenres(client, url, headers, preferences)
+        }
+        return super.fetchPopularManga(page)
+    }
+
+    // =============================== Latest ===============================
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        return GET("$baseUrl/comic/bookclass.html?type=last_release&page_num=$page&language=$lang", headers)
+    }
+
+    override fun latestUpdatesSelector(): String = popularMangaSelector()
+
+    override fun latestUpdatesFromElement(element: Element): SManga = popularMangaFromElement(element)
+
+    override fun latestUpdatesNextPageSelector(): String? = popularMangaNextPageSelector()
+
+    // =============================== Search ===============================
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        if (query.isNotEmpty()) {
+            return GET("$baseUrl/comic/search.html?keyword=$query&page_num=$page", headers)
+        }
+        filters.forEach { filter ->
+            when (filter) {
+                is GenreFilter -> {
+                    val filterId = filter.selected
+                    Log.i("NovelFullGlobal_Genre", "Filtering by genre: $filterId")
+                    return GET("$baseUrl/comic/bookclass.html?type=category_novel&id=$filterId&page_num=$page&language=$lang", headers)
+                }
+                else -> { }
+            }
+        }
+        return GET("$baseUrl/comic/search.html?keyword=", headers)
+    }
+
+    override fun searchMangaSelector(): String = popularMangaSelector()
+
+    override fun searchMangaFromElement(element: Element): SManga = popularMangaFromElement(element)
+
+    override fun searchMangaNextPageSelector(): String? = popularMangaNextPageSelector()
+
+    override fun fetchSearchManga(
+        page: Int,
+        query: String,
+        filters: FilterList,
+    ): Observable<MangasPage> {
+        if (!supportsSearch && query.isNotEmpty()) {
+            return Observable.just(MangasPage(emptyList(), false))
+        }
+        return super.fetchSearchManga(page, query, filters)
+    }
+
+    // =============================== Filters ==============================
+
+    override fun getFilterList() = FilterList(
+        Note,
+        Filter.Separator(),
+        getGenreFilterOrTip(preferences),
+    )
+
+    // =========================== Manga Details ============================
+
+    override fun mangaDetailsParse(document: Document): SManga {
+        return SManga.create().apply {
+            title = document.selectFirst("h3.title")!!.text()
+            description = document.selectFirst(".desc-text")!!.text()
+                .replace("\\n", "\n")
+            thumbnail_url = document.selectFirst(".books > .book > img")!!.imgAttr()
```

Comment:
Remove non-null from `description` and `thumbnail_url`, the extension can work without these set.